pipeline-runner/R/gem2s-7-upload_to_aws.R

```diff
@@ -144,16 +145,20 @@ meta_sets <- function(input, scdata, color_pool) {
 
     # values of current metadata track
     values <- unique(meta[[i]])
-    for (i in seq_along(values)) {
-      value <- values[i]
+
+    for (j in seq_along(values)) {
+      value <- values[j]
       cell_ids <- scdata$cells_id[scdata[[seurat_key]] == value]
 
-      cell_set$children[[i]] <- list(
+      cell_set$children[[j]] <- list(
         "key" = paste(key, value, sep = "-"),
         "name" = value,
-        "color" = color_pool[i],
```

Comment:
this was re-using the same colors because `i` would start from `1:length(values)` for each metadata column.